src/shillelagh/adapters/base.py

```diff
@@ -51,10 +53,62 @@ class Adapter:
     # if true, the requested columns will be passed to ``get_rows`` and ``get_data``
     supports_requested_columns = False
 
+    # if true, a set of strings from an in statement will be serialized before being
+    # passed in as bounds to ``get_rows`` and ``get_data``. This requires
+    # supports_requested_columns to be set to True as well.
+    supports_in_statements = False
+
     def __init__(self, *args: Any, **kwargs: Any):  # pylint: disable=unused-argument
         # ensure ``self.close`` gets called before GC
         atexit.register(self.close)
 
+    @staticmethod
```

Comment:
Why go through the trouble of custom serialization / deserialization logic for sets rather than just serializing as a list  (and then if need be creating a set from the list)?

Reply:
Because at the time when you're evaluating what to do with the set in get_data() or get_rows(), I found no way to distinguish the serialized content from a string, so just converting to a comma separated list looked dangerous to me.

Reply:
removed